.github/workflows/test.yml

```diff
@@ -8,182 +8,204 @@ on:
 
-  test-mac:
-    name: sbt test on mac
-    runs-on: macos-latest
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v2
-      - name: Setup Scala
-        uses: olafurpg/setup-scala@v10
-        with:
-          java-version: adopt@1.11
-      - name: Test
-        run: sbt "testOnly -- -l RequiresVcs -l RequiresVerilator -l Formal -l RequiresIcarus"
+  #test-mac:
+  #  name: sbt test on mac
+  #  runs-on: macos-latest
+  #  steps:
+  #    - name: Checkout
+  #      uses: actions/checkout@v2
+  #    - name: Setup Scala
+  #      uses: olafurpg/setup-scala@v10
+  #      with:
+  #        java-version: adopt@1.11
+  #    - name: Test
+  #      run: sbt "testOnly -- -l RequiresVcs -l RequiresVerilator -l Formal -l RequiresIcarus"
 
-  icarus:
-    name: icarus verilog
-    runs-on: ${{ matrix.os }}
-    strategy:
-      matrix:
-        os: [ ubuntu-20.04, macos-latest ]
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v2
-      - name: Install Icarus Verilog for Ubuntu
-        if: runner.os == 'Linux'
-        run: |
-          sudo apt-get install -y iverilog
-          iverilog -v || true
-      - name: Install Icarus Verilog for MacOS
-        if: runner.os == 'macOS'
-        run: |
-          brew install icarus-verilog
-          iverilog -v || true
-      - name: Setup Scala
-        uses: olafurpg/setup-scala@v10
-        with:
-          java-version: adopt@1.11
-      - name: Test
-        run: sbt ++${{ matrix.scala }} "testOnly -- -n RequiresIcarus"
+  #icarus:
+  #  name: icarus verilog
+  #  runs-on: ${{ matrix.os }}
+  #  strategy:
+  #    matrix:
+  #      os: [ ubuntu-20.04, macos-latest ]
+  #  steps:
+  #    - name: Checkout
+  #      uses: actions/checkout@v2
+  #    - name: Install Icarus Verilog for Ubuntu
+  #      if: runner.os == 'Linux'
+  #      run: |
+  #        sudo apt-get install -y iverilog
+  #        iverilog -v || true
+  #    - name: Install Icarus Verilog for MacOS
+  #      if: runner.os == 'macOS'
+  #      run: |
+  #        brew install icarus-verilog
+  #        iverilog -v || true
+  #    - name: Setup Scala
+  #      uses: olafurpg/setup-scala@v10
+  #      with:
+  #        java-version: adopt@1.11
+  #    - name: Test
+  #      run: sbt ++${{ matrix.scala }} "testOnly -- -n RequiresIcarus"
 
   verilator-mac:
     name: verilator regression on mac
     runs-on: macos-latest
     steps:
       - name: Checkout
         uses: actions/checkout@v2
+      - name: Install GNU binutils
+        run: |
+          brew install binutils
+          # Use GNU binutils over built-in binutils
+          echo "ldflags=-L/usr/local/opt/binutils/lib" >> $GITHUB_ENV
+          echo "cppflags=-I/usr/local/opt/binutils/include" >> $GITHUB_ENV
+          echo "/usr/local/opt/binutils/bin:$PATH" >> $GITHUB_PATH
       - name: Install Verilator
         run: |
-          brew install verilator
+          brew install verilator --build-from-source
           verilator --version
```

Comment:
I wouldn't do it this way long term, but yeah we might need to start building verilator from source just like we do for Linux since it seems that you need to use the GNU ar to compile Verilator itself for it to work right...

Anyway just trying out `brew --build-from-source` but assuming this works I'll make this build and cache the same way that the verilator regression do below.

Reply:
Is this going to be our recommendation for people who want to use chiseltest on Mac? How are you doing this on your own machine and would you mind adding some instructions to our Readme?

Reply:
> How are you doing this on your own machine

Honestly, I haven't been using Verilator on my on laptop lately because it's been broken due to the `ar` business. I need to figure out a reasonable way to get it working that I can encourage other people to emulate. It's pretty annoying that the version from package managers (both Homebrew and Macports) just don't actually work. 

Perhaps the real solution is to fix the packages such that they bring in GNU ar but I don't have much experience with that sort of thing...

Reply:
Is there a way to force brew to pick an older version of Verilator? Are there any older versions that are not broken?

Reply:
I think there is, trying it out.

Reply:
Okay the solution is:

```
brew extract --version=4.200 verilator homebrew/cask
brew install verilator@4.200
```

This is much better, although ideally the newest version of the package would work. It does build from source though which is a pain so need to figure out how to cache it in Github Actions